### PR TITLE
feat(rust): Add support for cargo-auditable binaries

### PR DESCRIPTION
**Pull request by tofay**
Per https://github.com/aquasecurity/trivy/issues/2664, https://github.com/rust-secure-code/cargo-auditable is a cargo subcommand to embed Rust crate dependency information into binaries for subsequent vulnerability checking.

This PR updates go-dep-parser so that it can read this dependency information, using https://github.com/microsoft/go-rustaudit - a module that extracts and parses the dependency information. 

Signed-off-by: Tom Fay <tomfay@microsoft.com>

### CODE REVIEW

**Review by knqyf263 — COMMENTED** [review]
Looks great! I left a small comment.

File: pkg/rust/binary/parse_test.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		},
+		{
+			name:      "sad path",
+			inputFile: "testdata/dummy",
+			wantErr:   "unrecognized executable format",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f, err := os.Open(tt.inputFile)
```
[the remark this answers is not in the document]
Comment (tofay) [reply pkg/rust/binary/parse_test.go:74]: added here, and to the golang binary tests which I based these off

[the remark this answers is not in the document]
Comment (knqyf263) [reply pkg/rust/binary/parse_test.go:74]: Thanks!

**Review by knqyf263 — APPROVED** [review]

**Comment by knqyf263** [comment]
Thanks for your great contribution!

**Pull request merged**